app/models/trip.rb

```diff
@@ -0,0 +1,15 @@
+class Trip < ApplicationRecord
+  belongs_to :driver
+  belongs_to :passenger
+
+  #helper method for create trip
+  def self.find_available_driver
+    drivers = Driver.all
+    driver = drivers.find_by(available?: true)
+    if driver
+      driver.update(available?: false)
+      driver.save
+      return driver
```

Comment:
Nice method! It has a lot of good logic in here. I've spoken to Chantelle about this, but it looks like the parts of logic that belong in `Trip` might remain as `find_available_driver`, but it might call some logic that lives in `Driver`... the bits about update and save in particular strike as "should belong in Driver", and may be called something like "become_unavailable"